pkg/legacy/tarball/write.go

```diff
@@ -214,6 +214,14 @@ func MultiWrite(refToImage map[name.Reference]v1.Image, w io.Writer) error {
 		if err != nil {
 			return err
 		}
+		history := cfg.History
+		if len(history) != 0 && len(layers) != len(history) {
+			return errors.Errorf("image config had layer history which did not match the number of layers, got len(history)=%d, len(layers)=%d, want len(history)=len(layers)", len(history), len(layers))
+		}
```

Comment:
NIt: If you else this off the if statement below, you can just say len(layers) != len(history).